src/pages/SignUp/signUp.test.tsx

```diff
@@ -0,0 +1,44 @@
+import { render, fireEvent, waitFor, screen } from '@testing-library/react';
+import SignUpForm from 'components/SignUpForm';
+import { useCreateUserMutation } from 'services/blackMarketApi';
+
+jest.mock('services/blackMarketApi', () => ({
+  useCreateUserMutation: jest.fn(),
+}));
+
+describe('SignUpForm', () => {
+  beforeEach(() => {
+    (useCreateUserMutation as jest.Mock).mockReturnValue([jest.fn(), { isSuccess: false }]);
+  });
+
+  test('renders the SignUpForm component', () => {
+    render(<SignUpForm />);
+    expect(screen.getByText('Sign Up')).toBeInTheDocument();
+  });
+
+  test('disables the Sign Up button when form is not filled', () => {
+    render(<SignUpForm />);
+    expect(screen.getByText('Sign Up')).toBeDisabled();
+  });
+
+  test('enables the Sign Up button when form is filled', async () => {
+    render(<SignUpForm />);
+    fireEvent.change(screen.getByPlaceholderText('Type your email'), {
+      target: { value: 'test@example.com' },
+    });
```

Comment:
What do you think about using `userEvents.type()` (https://testing-library.com/docs/user-event/utility/#type) instead of forcing the change event? IMO trying to simulate the user's behavior is a better approach to test functionality. Let me know your thoughts